checker/tests/resourceleak/RlcThisTest.java

```diff
@@ -0,0 +1,40 @@
+// @skip-test until the bug is fixed
+
+import org.checkerframework.checker.mustcall.qual.InheritableMustCall;
+import org.checkerframework.checker.mustcall.qual.MustCallAlias;
+import org.checkerframework.common.returnsreceiver.qual.This;
+
+class RlcThisTest {
+
+  @InheritableMustCall("a")
+  private class Foo {
+
+    void a() {}
+
+    @This Foo b1(Foo this) {
+      return this;
+    }
+
+    @MustCallAlias Foo b2(@MustCallAlias Foo this) {
+      return this;
+    }
+  }
+
+  void test() {
+    Foo f1 = new Foo();
+    f1.b1(); // RLC reports a FP at this line
+    f1.a();
+
+    Foo f2 = new Foo();
+    f2.b2(); // RLC reports a FP at this line
+    f2.a();
+
+    f1 = new Foo();
```

Comment:
For clarity I'd prefer this to assign a new variable `f3`.  Also don't we get a false positive warning for this resource?  If so we could maybe add a comment

Reply:
The error only occurs if there is a reassignment.  I have added versions of the test with and without the reassignment to show the difference in behavior.